sql/json-functions-generated-column.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Using TiDB as a Document Store
+category: user guide
+---
+
+# Using TiDB as a Document Store
+
+## About
+
+In order to functionally keep pace with the latest version of MySQL (version 5.7 and above) and provide better support for document storage business as well, TiDB supports JSON in the latest version. You can use JSON type fields in TiDB tables, and index the fields in the JSON document in the form of generated column. Based on this, you can flexibly deal with those business with uncertain schema, no longer limited by the bad read performance and lack of support for transaction in traditional document databases.
+
+## JSON Functions
+
+The support for JSON in TiDB mainly refers to the user interface of MySQL 5.7. For example, you can create a table that includes a JSON field to store complex information:
+
+```sql
+CREATE TABLE person （
+    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    address_info JSON
+）
+```
+
+When you insert data into a table, you can deal with those data with uncertain schema like this:
+
+```sql
+INSERT INTO person (name, address_info) VALUES ("John", '{"city": "Beijing"}');
+```
+
+Put a legal JSON string at the location corresponding to the JSON field, and then you can insert JSON into the table. TiDB will parse the text and saves it in a more compact and easy-to-access form.
+
+You can also convert other types of data into JSON using CAST:
+
+```sql
+INSERT INTO person (name, address_info) VALUES ("John", CAST('{"city": "Beijing"}' AS JSON));
+INSERT INTO person (name, address_info) VALUES ("John", CAST('123' AS JSON));
+INSERT INTO person (name, address_info) VALUES ("John", CAST(123 AS JSON));
+```
+
+Now, if you want to query all the users living in Beijing, what should you do? Is it necessary to pull back all the data and then filter in the business layer? Not really. The same with traditional document databases like MongoDB, TiDB supports complex combinations of query conditions in the user side.
+
+```sql
+SELECT id, name FROM person WHERE JSON_EXTRACT(address_info, '$.city') = 'Beijing');
+```
+
+`JSON_EXTRACT` is a function that TiDB supports. If you are familiar with JSON in MySQL 5.7, you must know it. The `JSON_EXTRACT` function in TiDB behaves the same as in MySQL. The meaning of the above function is to extract the `city` field from the `address_info` document. The second argument is a "path expression" and is used to specify which field to extract. The complete syntax description of the path expression is complex. Here are a few simple examples to help you understand its general use:
```

Comment:
TiDB supports the `JSON_EXTRACT` function which is exactly the same as in MySQL. The function is to extract the `city` field from the `address_info` document. The second argument is a "path expression" and is used to specify which field to extract. See the following few examples to help you understand the "path expression":